hclsyntax/expression_test.go

```diff
@@ -1156,7 +1156,22 @@ upper(
 			}),
 			1,
 		},
-
+		{ // splat with sensitive collection
+			`maps.*.enabled`,
+			&hcl.EvalContext{
+				Variables: map[string]cty.Value{
+					"maps": cty.ListVal([]cty.Value{
+						cty.MapVal(map[string]cty.Value{"enabled": cty.True}),
```

Comment:
I think splat in the scenario where a member is marked but the collection itself isn't impacted by this bug/change, but this could be a good opportunity to add that test coverage if not already present.